### DISCUSSION TITLE
Openapi.json does not support pydantic v1 and pydantic v2 at the same time

### DISCUSSION THREAD

**Discussion by ellyx13** [fastapi/fastapi#11516] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Union
from fastapi import FastAPI
from pydantic import v1 as pydantic_v1
from pydantic import BaseModel

app = FastAPI()

class Foo(pydantic_v1.BaseModel):
    message: str

class Bar(BaseModel):
    message: str



@app.post("/foo")
def read_item(data: Foo):
    data = data.dict()
    return {"message": data['message']}

@app.post("/bar")
def read_item(data: Bar):
    data = data.dict()
    return {"message": data['message']}

# Run the server with: uvicorn main:app --reload
```

### Description

I am using FastAPI 0.110.3 and pydantic v2.7.1. Previously I used pydantic v1 and am planning to migrate to pydantic v2. But I have a problem that FastAPI cannot declare 2 api at the same time using BaseModel of pydantic v1 and pydantic v2.

I did some research and found out that pydantic v1 uses openapi.json 3.0 and pydantic v2 uses openapi.json 3.1, so I think the two versions of openapi.json conflict with each other.

I want fastapi to load the api using pydantic v1 in /v1/api path. And the api using pydantic v2 will be in the v2/api path. But I get error

```
raise PydanticInvalidForJsonSchema(f'Cannot generate a JsonSchema for {error_info}')
pydantic.errors.PydanticInvalidForJsonSchema: Cannot generate a JsonSchema for core_schema.PlainValidatorFunctionSchema ({'type': 'with-info', 'function': <bound method BaseModel.validate of <class 'main.Foo'>>})
```

### Operating System

Linux

### Operating System Details

pydantic version: 2.7.1
pydantic-core version: 2.18.2
pydantic-core build: profile=release pgo=true
install path: /home/kietpham/python/pydantic/venv/lib/python3.10/site-packages/pydantic
python version: 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
platform: Linux-5.15.146.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
related packages: fastapi-0.110.3 typing_extensions-4.11.0
commit: unknown

### FastAPI Version

0.110.3

### Pydantic Version

2.7.1

### Python Version

Python 3.12.3

### Additional Context

Library version from pip freeze:
annotated-types==0.6.0
anyio==4.3.0
click==8.1.7
exceptiongroup==1.2.1
fastapi==0.110.3
h11==0.14.0
httptools==0.6.1
idna==3.7
pydantic==2.7.1
pydantic_core==2.18.2
python-dotenv==1.0.1
PyYAML==6.0.1
sniffio==1.3.1
starlette==0.37.2
typing_extensions==4.11.0
uvicorn==0.29.0
uvloop==0.19.0
watchfiles==0.21.0
websockets==12.0

**Comment by kyogom** [comment]
Hi, ellyx13.
I reproduced your problem.

> I did some research and found out that pydantic v1 uses openapi.json 3.0 and pydantic v2 uses openapi.json 3.1, so I think the two versions of openapi.json conflict with each other.

And I set openapi version [by extending openapi](https://fastapi.tiangolo.com/how-to/extending-openapi/#generate-the-openapi-schema) but it doesn't fix error. 
```
....

@app.post("/bar")
def read_item(data: Bar):
    data = data.dict()
    return {"message": data["message"]}


def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title="title",
        version="1.0.0",
        openapi_version="3.0.0", # set to 3.0.0
        routes=app.routes,
    )
    app.openapi_schema = openapi_schema
    return app.openapi_schema


app.openapi = custom_openapi
```

Fastapi seems refer pydantic.version and currently it doesn't seem possible to create openapi document using pydantic model v1 and v2 at the same time. 
https://github.com/tiangolo/fastapi/blob/e04d397e3224d3674cf73332303b956fc3d00b3d/fastapi/_compat.py#L23

**Comment by YuriiMotov** [comment]
Using Pydantic V1 and V2 in one FastAPI project is not possible for now.
And, I think this support will never be added, considering the expected Pydantic V3 release this year that will mean stopping Pydantic V1 version's maintenance.

Consider refactoring your project to use Pydantic V2.
